freeipa/src/main/java/com/sequenceiq/freeipa/service/recipe/RecipeUsageService.java

```diff
@@ -0,0 +1,54 @@
+package com.sequenceiq.freeipa.service.recipe;
+
+import javax.inject.Inject;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Service;
+
+import com.cloudera.thunderhead.service.common.usage.UsageProto;
+import com.sequenceiq.cloudbreak.usage.UsageReporter;
+
+@Service
+public class RecipeUsageService {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(RecipeUsageService.class);
+
+    @Inject
+    private UsageReporter usageReporter;
+
+    public void sendDetachedUsage(String stackCrn, String recipeName) {
+        try {
+            usageReporter.cdpRecipeEvent(UsageProto.CDPRecipeEvent.newBuilder()
+                    .setName(recipeName)
+                    .setStackCrn(stackCrn)
+                    .setStatus(UsageProto.CDPRecipeStatus.Value.DETACHED)
+                    .build());
+        } catch (Exception e) {
+            LOGGER.error("Couldn't send usage report about recipe detachment, because: ", e);
+        }
+    }
+
+    public void sendAttachedUsage(String stackCrn, String recipeName) {
+        try {
+            usageReporter.cdpRecipeEvent(UsageProto.CDPRecipeEvent.newBuilder()
+                    .setName(recipeName)
+                    .setStackCrn(stackCrn)
+                    .setStatus(UsageProto.CDPRecipeStatus.Value.ATTACHED)
+                    .build());
+        } catch (Exception e) {
+            LOGGER.error("Couldn't send usage report about recipe attachment, because: ", e);
+        }
+    }
+
+    public void sendClusterCreationRecipeUsage(String stackCrn, int recipeCount) {
+        try {
+            usageReporter.cdpClusterCreationRecipeEvent(UsageProto.CDPClusterCreationRecipeEvent.newBuilder()
+                    .setStackCrn(stackCrn)
+                    .setRecipeCount(recipeCount)
+                    .build());
+        } catch (Exception e) {
+            LOGGER.error("Couldn't send usage report about recipe creation, because: ", e);
```

Comment:
Same comment as before, it's not about recipe creation, but cluster creation with attached recipe.